codegenerator/cli/templates/static/codegen/src/Utils.res

```diff
@@ -132,6 +132,31 @@ Helper to check if a value exists in an array
       [...head, ...tail]
     }
   }
+
+  let last = (arr: array<'a>): option<'a> => arr->Belt.Array.get(arr->Array.length - 1)
+
+  let findReverseWithIndex = (arr: array<'a>, fn: 'a => bool): option<('a, int)> => {
+    let rec loop = (index: int) => {
+      if index < 0 {
+        None
+      } else {
+        let item = arr[index]
+        if fn(item) {
+          Some((item, index))
+        } else {
+          loop(index - 1)
+        }
+      }
+    }
+    loop(arr->Array.length - 1)
+  }
+
+  /** 
+  Currently a bug in rescript if you ignore the return value of spliceInPlace 
+  https://github.com/rescript-lang/rescript-compiler/issues/6991
+  */
+  @send
+  external spliceInPlace: (array<'a>, ~pos: int, ~remove: int) => array<'a> = "splice"
```

Comment:
Also a bug in the compiler that isn't solved by upgrade: https://github.com/rescript-lang/rescript-compiler/issues/6991